### ISSUE TITLE
[REQUEST] SRV DNS records support.

### ISSUE THREAD

**Issue by JabLuszko** [ValveSoftware/halflife#193]
I guess it was mentioned few times on mailling list already, but hey, this is the new tracker :-)

It would be nice if Steam would support SRV records. It already supports A records, so I guess it shouldn't be hard.

Pros:
- transparent for end-user. They add to favourites/connect to same canonical name every time. They don't care if server moved to other GSP, changed port.
- easier to remember address - we both know that "official.valve.com" is much easier than 178.27.237.27:27604
- I guess Valve could use it for some kind of load-balancing between theirs servers.

Cons: 
- more DNS request.

[1] http://tools.ietf.org/html/rfc2782
[2] http://en.wikipedia.org/wiki/SRV_record

**Comment by ztjuh** [comment]
Any progress on this? I don't think we will ever get this! :(

**Comment by zharding** [comment]
This is actually a really good suggestion. I know, I am literally years behind lol. 

The current workaround is to use iptables in linux. SRV records would be by far more stable and appropriate. I mean SRV records exist for a reason...

In response to @yaakov-h : 
SRV records really dont work that way. I mean they do... However, they are intended to direct traffice for a service to the right host and port. So, multiple entries for the same result host, protocol, and port would be pointless. If you chose to load balance, pick a game that supports load balancing like Minecraft with BungeeCord? 

**SRV Record Format**
`_service._tcp.my.server.net.   TTL IN    SRV P       W     PORT final.host.net.`

**Proposed SRV Record format for Valve**
`_hl2-[gameid]._tcp.my.server.net.   3600  IN    SRV 0       5     27015 my.server.net.`

This is a necessary for providers. Enthusiasts would benefit as well.

**Comment by RoofLee** [comment]
> This is actually a really good suggestion. I know, I am literally years behind lol.
> 
> The current workaround is to use iptables in linux. SRV records would be by far more stable and appropriate. I mean SRV records exist for a reason...
> 
> In response to @yaakov-h :
> SRV records really dont work that way. I mean they do... However, they are intended to direct traffice for a service to the right host and port. So, multiple entries for the same result host, protocol, and port would be pointless. If you chose to load balance, pick a game that supports load balancing like Minecraft with BungeeCord?
> 
> **SRV Record Format**
> `_service._tcp.my.server.net. TTL IN SRV P W PORT final.host.net.`
> 
> **Proposed SRV Record format for Valve**
> `_hl2-[gameid]._tcp.my.server.net. 3600 IN SRV 0 5 27015 my.server.net.`
> 
> This is a necessary for providers. Enthusiasts would benefit as well.

How I can make a SRV record for garrysmod? Will this work?

**Comment by nikrozman21** [comment]
Any updates on this ?

**Comment by realsnowy** [comment]
We need this, this would be so handy to protect your server with cloudflare to prevent DDoS attacks or to hide your IP if you wouldn't want that public. Just saying... :^)

**Comment by nikrozman21** [comment]
In reply to realsnowy you can't really hide your IP with SRV records. They still have to be pointing to an "open" subdomain. You can't utilize Cloudflare for that task cause they don't proxy on any other ports than just 80 and 443 as far as I'm aware.

**Comment by realsnowy** [comment]
I can still put my Minecraft server behind cloudflare and it works. You have to make a proxied A record and make the SRV record point to that proxied domain.

**Comment by burakataoglu** [comment]
any update ?

**Comment by oaus** [comment]
@mikela-valve Is this possible in a future version?

**Comment by bcdoor** [comment]
@zharding I agree. Do you have any idea on how to configure the iptables as this issue is still not fixed.. Would really appreciate the help!

**Comment by afwn90cj93201nixr2e1re** [comment]
It's not nice idea, better to use steam server's api, which can update ip's just by server id and owner account.

Domain control can be taken by intruders.

So, we should just resolve server ip on server adding step, and save it as plain ipv4.

**Comment by Kourgiantakis** [comment]
I'm shocked that this is still not supported.

**Comment by jsebean** [comment]
I am too.

**Comment by loriopatrick** [comment]
I would really like this feature. I've been working on a tunneling service for games and this would allow me to provide a much better experience for my users.

**Comment by fabiocasolari** [comment]
i thought Valve response to this is....   go play fortnite fuckers...

**Comment by kubo6472** [comment]
Anything new?

**Comment by Be1zebub** [comment]
any progress?

**Comment by kubo6472** [comment]
probably not

**Comment by datemike972** [comment]
valve, please

**Comment by Be1zebub** [comment]
> This is actually a really good suggestion. I know, I am literally years behind lol.
> 
> The current workaround is to use iptables in linux. SRV records would be by far more stable and appropriate. I mean SRV records exist for a reason...
> 
> In response to @yaakov-h : SRV records really dont work that way. I mean they do... However, they are intended to direct traffice for a service to the right host and port. So, multiple entries for the same result host, protocol, and port would be pointless. If you chose to load balance, pick a game that supports load balancing like Minecraft with BungeeCord?
> 
> **SRV Record Format** `_service._tcp.my.server.net. TTL IN SRV P W PORT final.host.net.`
> 
> **Proposed SRV Record format for Valve** `_hl2-[gameid]._tcp.my.server.net. 3600 IN SRV 0 5 27015 my.server.net.`
> 
> This is a necessary for providers. Enthusiasts would benefit as well.

iptables? please tell me how 🙏

**Comment by R4to0** [comment]
Instead of a SRV DNS record, why not tie the gameserver with a **[Steam Game Server Account Management](https://steamcommunity.com/dev/managegameservers)** and update the IP address using the static Steam Gameserver ID? That way DNS wouldn't be needed and Steam could update the IP address entry by that ID. [That's how Garry's Mod does actually!](https://wiki.facepunch.com/gmod/Steam_Game_Server_Accounts)

**Comment by Sammot** [comment]
Please can we have this

**Comment by sphaxa** [comment]
bump please this would make our lives so much easier when using services like dathost, which so many services do.

**Comment by ztjuh** [comment]
Too hard for Valve 😂

**Comment by EzioTheDeadPoet** [comment]
This issue is now 11 years old 😭

**Comment by 2lag** [comment]
would they accept a pr to add it 😭

**Comment by prljav** [comment]
> This issue is now 11 years old 😭

4,188 days to be exact

**Comment by Lunascaped** [comment]
2025 and still nothin bruh

**Comment by ztjuh** [comment]
😂

**Comment by ztjuh** [comment]
i'M BORED
